### PR TITLE
Support clang-format whose version is not 7.0

### PR DESCRIPTION
**Pull request by jiangzihao2009**
## What do these changes do?
Print a warning and ask user whether to continue when installed version of clang-format is not 7.0.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: ci/travis/format.sh [hunk 1]
```diff
@@ -38,8 +38,11 @@ fi
 if which clang-format >/dev/null; then
   CLANG_FORMAT_VERSION=$(clang-format --version | awk '{print $3}')
   if [[ $CLANG_FORMAT_VERSION != "7.0.0" ]]; then
-    echo "Please install clang-format 7.0.0. You currently are using $CLANG_FORMAT_VERSION."
-    exit 1
+    echo "WARNING: Please install clang-format 7.0.0. You currently are using $CLANG_FORMAT_VERSION."
+    read -p "Are you sure to continue?[y/n]" input
```
Comment (raulchen) [inline ci/travis/format.sh]: Maybe rephrase like this?
```
WARNING: Ray uses clang-format 7.0.0, but you are using $CLANG_FORMAT_VERSION. This might generate different results. Do you want to continue? [y/n]
```

Comment (raulchen) [reply ci/travis/format.sh]: And we should do the same thing for other tools as well.

File: ci/travis/format.sh [hunk 2]
```diff
@@ -26,23 +26,36 @@ FLAKE8_VERSION=$(flake8 --version | awk '{print $1}')
...
+         "This might generate different results."
+    read -p "Do you want to continue? [y/n]" input
+    if ! [ $input = 'y' ]; then
+        exit 1
+    fi
   fi
 else
-  echo "WARNING: clang-format is not installed!"
+    echo "WARNING: clang-format is not installed!"
+    exit 1
```
Comment (robertnishihara) [inline ci/travis/format.sh]: I didn't exit here before because I thought some Python developers (e.g., for RLlib) might not install `clang-format`. It might be better to warn them and then ask if they want to proceed?

**Review by robertnishihara — COMMENTED** [review]
This looks good to me! I left one comment that I think should be addressed.

**Review by robertnishihara — APPROVED** [review]

**Review by raulchen — APPROVED** [review]
